src/cdk/drag-drop/drop-list.ts

```diff
@@ -104,21 +105,30 @@ export class CdkDropList<T = any> implements OnInit, OnDestroy {
   constructor(
     public element: ElementRef<HTMLElement>,
     private _dragDropRegistry: DragDropRegistry<CdkDrag, CdkDropList<T>>,
-    @Optional() private _dir?: Directionality) {}
+    @Optional() private _dir?: Directionality,
+    @Optional() private _group?: CdkDropListGroup<CdkDropList>) {}
 
   ngOnInit() {
     this._dragDropRegistry.registerDropContainer(this);
+
+    if (this._group) {
+      this._group._items.add(this);
```

Comment:
I was debating whether this should live in the `CdkDropRegistry`. I decided not to do it, because it was adding a fair bit of code for no real benefit.